libraries/Microsoft.Bot.Builder.Azure.Blobs/BlobsTranscriptStore.cs

```diff
@@ -406,7 +421,23 @@ private async Task LogActivityToBlobClientAsync(IActivity activity, BlobClient b
                 ["RecipientId"] = activity.Recipient?.Id,
                 ["Timestamp"] = activity.Timestamp.Value.ToString("O", CultureInfo.InvariantCulture)
             };
-            await blobClient.SetMetadataAsync(metaData).ConfigureAwait(false);
+
+            var options = new BlobUploadOptions
+            {
+                Metadata = metaData,
+                TransferOptions = _storageTransferOptions
```

Comment:
What happens if this is null? Does blob sdk complain?

Reply:
I don't use .NET much, but it's non-nullable and I can't seem to pass in `null` to either the constructor or to `options`. Should I add this in as a safety mechanism, anyway?

```csharp
if (storageTransferOptions == null)
{
    throw new ArgumentNullException(nameof(storageTransferOptions));
}
```

Reply:
You could write a test with null options and verify that it works. If that is the case, we should be fine. If the client complains, we could provide a reasonable default if no options are provided.